unable to detect platform for scan container registry

```bash
cnspec scan container registry docker.io/pmuench/dvwa-container-escape --score-threshold 25 --output unit 

→ loaded configuration from /Users/atomic111/.config/mondoo/mondoo.yml using source default
→ using service account credentials
→ discover related assets for 1 asset(s)
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x98 pc=0x101bfa08c]

goroutine 1 [running]:
go.mondoo.com/cnspec/v9/policy/scan.(*LocalScanner).distributeJob(0x1400025e620, 0x1400025e7e0, {0x10221c3f8, 0x140008f6d20}, 0x140007a8180)
	/Users/atomic111/go/src/go.mondoo.com/cnspec/policy/scan/local_scanner.go:338 +0xb2c
go.mondoo.com/cnspec/v9/policy/scan.(*LocalScanner).Run(0x1400025e620, {0x10221c3f8, 0x140008f6d20}, 0x1400025e7e0)
	/Users/atomic111/go/src/go.mondoo.com/cnspec/policy/scan/local_scanner.go:158 +0x118
go.mondoo.com/cnspec/v9/apps/cnspec/cmd.RunScan(0x140008ec000, {0x0?, 0x0?, 0x0?})
	/Users/atomic111/go/src/go.mondoo.com/cnspec/apps/cnspec/cmd/scan.go:352 +0x380
go.mondoo.com/cnspec/v9/apps/cnspec/cmd.glob..func13(0x102ca04e0?, 0x140001d43c0?, 0x102cd6740?)
	/Users/atomic111/go/src/go.mondoo.com/cnspec/apps/cnspec/cmd/scan.go:131 +0xa8
go.mondoo.com/cnquery/v9/cli/providers.setConnector.func2(0x140007c8a00?, {0x1400065b920?, 0x2, 0x6})
	/Users/atomic111/go/src/go.mondoo.com/cnquery/cli/providers/providers.go:490 +0x898
github.com/spf13/cobra.(*Command).execute(0x140007ec600, {0x1400065b8c0, 0x6, 0x6})
	/Users/atomic111/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944 +0x640
github.com/spf13/cobra.(*Command).ExecuteC(0x102ba5900)
	/Users/atomic111/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x320
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/atomic111/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
go.mondoo.com/cnspec/v9/apps/cnspec/cmd.Execute()
	/Users/atomic111/go/src/go.mondoo.com/cnspec/apps/cnspec/cmd/root.go:114 +0x19c
main.main()
	/Users/atomic111/go/src/go.mondoo.com/cnspec/apps/cnspec/cnspec.go:9 +0x1c
atomic111:../workspace-waldner ±>                                                                                                                                                               x2 1h57m main[0587847]
```

```bash
cnspec scan container docker.io/pmuench/dvwa-container-escape --score-threshold 25 --output junit                                                           x130 1h57m main[0587847]
→ loaded configuration from /Users/atomic111/.config/mondoo/mondoo.yml using source default
→ using service account credentials
→ discover related assets for 1 asset(s)
x unable to connect to asset error="rpc error: code = Unknown desc = Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x40 pc=0x10129e3b4]

goroutine 1 [running]:
go.mondoo.com/cnspec/v9/cli/reporter.ReportCollectionToJunit(0x0, {0x1018f3938, 0x140005ba7c0})
	/Users/atomic111/go/src/go.mondoo.com/cnspec/cli/reporter/junit.go:27 +0x44
go.mondoo.com/cnspec/v9/cli/reporter.(*Reporter).Print(0x16f7e7514?, 0x0, {0x1018e8048, 0x14000078038})
	/Users/atomic111/go/src/go.mondoo.com/cnspec/cli/reporter/reporter.go:159 +0x2bc
go.mondoo.com/cnspec/v9/apps/cnspec/cmd.printReports(0x10130b2fa?, 0x14000981400, 0x10183d4a0?)
	/Users/atomic111/go/src/go.mondoo.com/cnspec/apps/cnspec/cmd/scan.go:376 +0xa4
go.mondoo.com/cnspec/v9/apps/cnspec/cmd.glob..func13(0x1023804e0?, 0x14000188500?, 0x1023b6740?)
	/Users/atomic111/go/src/go.mondoo.com/cnspec/apps/cnspec/cmd/scan.go:137 +0x10c
go.mondoo.com/cnquery/v9/cli/providers.setConnector.func2(0x14000679400?, {0x140008062d0?, 0x1, 0x5})
	/Users/atomic111/go/src/go.mondoo.com/cnquery/cli/providers/providers.go:490 +0x898
github.com/spf13/cobra.(*Command).execute(0x140006ca300, {0x14000806280, 0x5, 0x5})
	/Users/atomic111/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:944 +0x640
github.com/spf13/cobra.(*Command).ExecuteC(0x102285900)
	/Users/atomic111/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:1068 +0x320
github.com/spf13/cobra.(*Command).Execute(...)
	/Users/atomic111/go/pkg/mod/github.com/spf13/cobra@v1.7.0/command.go:992
go.mondoo.com/cnspec/v9/apps/cnspec/cmd.Execute()
	/Users/atomic111/go/src/go.mondoo.com/cnspec/apps/cnspec/cmd/root.go:114 +0x19c
main.main()
	/Users/atomic111/go/src/go.mondoo.com/cnspec/apps/cnspec/cnspec.go:9 +0x1c
```

--- a/explorer/scan/local_scanner.go
+++ b/explorer/scan/local_scanner.go
@@ -267,17 +267,21 @@ func (s *LocalScanner) distributeJob(job *Job, ctx context.Context, upstream *up
 			}
 		}
 		runtime.SetRecording(candidate.runtime.Recording)
-
 		err = runtime.Connect(&plugin.ConnectReq{
 			Features: config.Features,
 			Asset:    candidate.asset,
 			Upstream: upstream,
 		})
+
 		if err != nil {
 			log.Error().Err(err).Str("asset", candidate.asset.Name).Msg("unable to connect to asset")
 			continue
 		}
 
+		if candidate.asset.GetPlatform() == nil {
+			log.Error().Msgf("unable to detect platform for asset " + candidate.asset.Name)
+			continue
+		}
 		assets = append(assets, &assetWithRuntime{
 			asset:   candidate.asset,
 			runtime: runtime,

--- a/providers/os/provider/provider.go
+++ b/providers/os/provider/provider.go
@@ -558,6 +558,10 @@ func (s *Service) discoverRegistry(conn *connection.TarConnection) (*inventory.I
 	}
 
 	inventory := &inventory.Inventory{}
+	// we detect the platform for each asset we discover here
+	for _, a := range resolvedAssets {
+		_ = s.detect(a, conn)
+	}
 	inventory.AddAssets(resolvedAssets...)
 
 	return inventory, nil

--- a/providers/os/provider/provider.go
+++ b/providers/os/provider/provider.go
@@ -560,6 +560,7 @@ func (s *Service) discoverRegistry(conn *connection.TarConnection) (*inventory.I
 	inventory := &inventory.Inventory{}
 	// we detect the platform for each asset we discover here
 	for _, a := range resolvedAssets {
+		// ignore the error. we will retry detection if we connect to the asset
 		_ = s.detect(a, conn)
 	}
 	inventory.AddAssets(resolvedAssets...)

